examples/training/semantic_segmentation/pascal_voc/deeplab_v3.py

```diff
@@ -129,12 +136,13 @@ def proc_eval_fn(examples):
         boundaries=[30000 * 16 / global_batch],
         values=[base_lr, 0.1 * base_lr],
     )
-    model = DeepLabV3(
-        classes=21,
-        backbone="resnet50_v2",
+    backbone = ResNet50V2(
         include_rescaling=True,
-        backbone_weights="imagenet",
+        input_shape=(512, 512, 3),
+        include_top=False,
+        weights="imagenet",
     )
+    model = DeepLabV3(classes=21, backbone=backbone, input_shape=(512, 512, 3))
```

Comment:
nit -- can DeepLab use `input_shape=None` by default and infer the input shape from backbone, but error out if neither `input_shape` is provided or backbone input_shape is also unknown?

Reply:
Done